docs/common-problems.asciidoc

```diff
@@ -240,3 +241,29 @@ For example:
 APM Agent --> Load Balancer  --> APM Server 
    10s            15s               30s
 ----------------------------------------------------------------------
+
+[[server-es-down]]
+[float]
+=== What happens to data when APM Server or Elasticsearch is down?
+
+==== Elasticsearch down
+
+If Elasticsearch goes down, the APM Server will keep data in memory until Elasticsearch is back up,
+or until it runs out of space in its internal in-memory queue. 
+You can <<adjust-queue-size,adjust the internal queue size>> if necessary.
+When the queue becomes full, APM Server will respond with an <<queue-full`HTTP 503: Queue is full`>>,
+and data will be lost.
+
+==== APM Server down
+
+Some agents have internal queues or buffers to temporarily store data in case the APM Server goes down.
+As a general rule of thumb, queues fill up quickly. Assume data will be lost if APM Server goes down.
+Adjusting these queues/buffers can increase the overhead of the agent, so use caution when updating default values.
+
+* Go Agent - `ELASTIC_APM_BUFFER_SIZE`/`ELASTIC_APM_REQUEST_SIZE`?
+* Java Agent - {}/config-reporter.html#config-max-queue-size[`max_queue_size`]
+* Node.js Agent - No internal queue. Data is lost
+* Python Agent - {}/tuning-and-overhead.html#tuning-queue[Transaction Queue]
+* Ruby Agent - {}/configuration.html#config-api-buffer-size[`api_buffer_size`]
+* RUM Agent - ?
```

Comment:
@jahtalab / @vigneshshanmugam - can one of you help me with this?

Reply:
@bmorelli25,  We don't retry on failures so if the APM server is down the data is lost. However, if APM server is up but ES is down then APM server will keep events in a queue until that queue becomes full and then starts with request failures and from that point on the data is lost.

Reply:
Thank you Hamid!